compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5839,16 +5844,22 @@ TR::Register * OMR::Power::TreeEvaluator::ibyteswapEvaluator(TR::Node *node, TR:
          {
          generateTrg1Src1Instruction(cg, TR::InstOpCode::brw, node, tgtRegister, srcRegister);
          }
+      else if (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P8))
+         {
+         generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rlwimi, node, tgtRegister, srcRegister, 24, CONSTANT64(0x00ffffff00));
```

Comment:
Why is `rlwimi` used for the first write to `tgtRegister`? All this seems to do is create a false dependency on the old value in `tgtRegister`. Why not just use `rlwinm` to eliminate that?

The same issue also appears in the other evaluators.

Reply:
I think the argument for `rlwimi` was that in P8 it's 1 cycle while `rlwinm` is 1-2 cycles. But I agree that the false dependency could cause hardware waiting for the value in `tgtRegister` if its not ready even though we don't need it. Will change it.

Reply:
Fixed in bb0c249